### PR TITLE
Add backend validation while adding a invoice

### PR DESCRIPTION
**Pull request by Ayushigupta2**
Target #1939

### Description
 I have added backend validation on all the data coming in the  add new invoice form.
 Applied old function on all the data in the form 

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Invoice/Http/Controllers/InvoiceController.php [hunk 1]
```diff
@@ -69,9 +70,9 @@ public function monthlyGSTTaxReportExport(Request $request)
 
     /**
      * Store a newly created resource in storage.
-     * @param Request $request
+     * @param \Modules\Invoice\Http\Requests\StorePostRequest $request
```
Comment (GauravGusain98) [inline Modules/Invoice/Http/Controllers/InvoiceController.php:73]:
```suggestion
     * @param StorePostRequest $request
```

File: Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php [hunk 2]
```diff
@@ -62,15 +62,15 @@ class="ml-2"
                 </div>
                 <div class="form-group ">
                     <label for="comments">Comments</label>
-                    <textarea name="comments" id="comments" rows="5" class="form-control"></textarea>
+                    <textarea name="comments" id="comments" rows="5" class="form-control">{{old('comments') }}</textarea>
```
Comment (GauravGusain98) [inline Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php:65]:
```suggestion
                    <textarea name="comments" id="comments" rows="5" class="form-control">{{ old('comments') }}</textarea>
```

File: Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php [hunk 3]
```diff
@@ -62,15 +62,15 @@ class="ml-2"
...
                 </div>
             </div>
             <div class="col-md-5 offset-md-1">
                 <div class="form-group">
                     <label for="project_invoice_id" class="field-required">Status</label>
                     <select class="form-control" name="status">
-                        <option value="sent">Sent</option>
-                        <option value="paid">Paid</option>
+                        <option value="sent" {{ old('status')=="sent" ? "selected" : ''}}>Sent</option>
```
Comment (GauravGusain98) [inline Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php:72]:
```suggestion
                        <option value="sent" {{ old('status')=="sent" ? "selected" : '' }}>Sent</option>
```

File: Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php [hunk 4]
```diff
@@ -62,15 +62,15 @@ class="ml-2"
...
+                        <option value="paid" {{ old('status')=="paid" ? "selected" : ''}}>Paid</option>
```
Comment (GauravGusain98) [inline Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php:73]:
```suggestion
                        <option value="paid" {{ old('status')=="paid" ? "selected" : '' }}>Paid</option>
```

File: Modules/Invoice/Http/Requests/StorePostRequest.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            'currency' => 'required',
+            'comments' => 'required',
+            'amount' => 'required',
+            'gst' =>'required',
+            'term' => 'required',
+            'sent_on' => 'required',
+            'due_on' => 'required',
+            'invoice_file' =>'required',
+
+        ];
```
Comment (GauravGusain98) [inline Modules/Invoice/Http/Requests/StorePostRequest.php:29]:
```suggestion
        $rules = [
            'client_id' =>'required|integer',
            'billing_level' => 'required|string',
            'status' => 'required|string',
            'currency' => 'required|string',
            'comments' => nullable|string",
            'amount' => 'required|numeric',
            'gst' =>'required|numeric',
            'term' => 'required',
            'sent_on' => 'required|date',
            'due_on' => 'required|date',
            'invoice_file' =>'required|mimes:pdf',
        ];
```

File: Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php [hunk 6]
```diff
@@ -108,12 +108,12 @@ class="ml-2"
                 <div class="form-group">
                     <label for="sent_on" class="field-required">Sent on</label>
                     <input type="date" class="form-control" name="sent_on" id="sent_on" required="required"
-                        value="{{ now()->format('Y-m-d') }}">
+                        value="{{ old('sent_on', now()->format('Y-m-d'))}}">
```
Comment (GauravGusain98) [inline Modules/Invoice/Resources/views/subviews/create/invoice-details.blade.php:111]:
```suggestion
                        value="{{ old('sent_on', now()->format('Y-m-d')) }}">
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please fix merge conflicts and check the comment

File: Modules/Invoice/Http/Requests/StorePostRequest.php (hunk 5 above)
Comment (Ayushigupta2) [reply Modules/Invoice/Http/Requests/StorePostRequest.php]: Done with suggested changes

File: Modules/Invoice/Http/Requests/StorePostRequest.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            'status' => 'required',
+            'currency' => 'required',
+            'comments' => 'required',
+            'amount' => 'required',
+            'gst' =>'required',
+            'term' => 'required',
+            'sent_on' => 'required',
+            'due_on' => 'required',
+            'invoice_file' =>'required',
+        ];
```
Comment (GauravGusain98) [inline Modules/Invoice/Http/Requests/StorePostRequest.php:28]: The changes requested earlier to this file is not done.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please make the required changes and fix merge conflicts

File: Modules/Invoice/Http/Requests/StorePostRequest.php (hunk 7 above)
Comment (Ayushigupta2) [reply Modules/Invoice/Http/Requests/StorePostRequest.php]: done with these changes now
i am getting problem in resolving conflicts i have take pull from master and accept both changes and then commited there only and changes total 190 files so i reset commit to old commit.

File: Modules/Invoice/Http/Controllers/InvoiceController.php [hunk 8]
```diff
@@ -72,9 +73,9 @@ public function monthlyGstTaxReportExport(Request $request)
 
     /**
      * Store a newly created resource in storage.
-     * @param Request $request
+     * @param StorePostRequest $request
      */
-    public function store(Request $request)
+    public function store(StorePostRequest $request)
```
Comment (GauravGusain98) [inline Modules/Invoice/Http/Controllers/InvoiceController.php:78]: Why we are using `StorePostRequest`? What does it do?

**Pull request closed without merging**